Formula/step.rb

```diff
@@ -0,0 +1,45 @@
+class Step < Formula
+  desc "Crypto and x509 Swiss-Army-Knife"
+  homepage "https://smallstep.com"
+  url "https://github.com/smallstep/cli/releases/download/v0.8.3/brew_step_0.8.3.tar.gz"
+  sha256 "6f93906ff1bea17dad02e8509fd06351ff3d8bdaa921b46f36aacfa47c44c762"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/smallstep").install buildpath.children
+    cd "src/github.com/smallstep/cli" do
+      system "make", "build"
+      bin.install "bin/step" => "step"
+      bash_completion.install "autocomplete/bash_autocomplete" => "step"
+    end
+    cd "src/github.com/smallstep/certificates" do
+      system "make", "build"
+      bin.install "bin/step-ca" => "step-ca"
+    end
+  end
+
+  test do
+    # Generate a public / private key pair. Creates foo.pub and foo.priv.
+    system "#{bin}/step", "crypto", "keypair", "foo.pub", "foo.priv", "--no-password", "--insecure"
+    assert_equal "foo.priv\nfoo.pub\n", shell_output("ls foo*")
+
+    # Generate a root certificate and private key with subject baz written to baz.crt and baz.key.
+    system "#{bin}/step", "certificate", "create", "--profile", "root-ca", "--no-password", "--insecure", "baz", "baz.crt", "baz.key"
+    assert_equal "baz.crt\nbaz.key\n", shell_output("ls baz*")
+    assert_equal "Subject: CN=baz\n", shell_output("#{bin}/step certificate inspect baz.crt | grep 'Subject: CN' | sed -e 's/^[ \t]*//'")
+    assert_equal "Issuer: CN=baz\n", shell_output("#{bin}/step certificate inspect baz.crt | grep 'Issuer: CN' | sed -e 's/^[ \t]*//'")
```

Comment:
Again, we shouldn't be shelling out & firing up unneeded processes/relying on the behaviour of `grep` and `sed`. Instead can you please use `assert_match` in combination with `shell_output()`? You can look at [avimetaedit.rb L32](https://github.com/Homebrew/homebrew-core/blob/master/Formula/avimetaedit.rb#L32) or [fribidi.rf L29](https://github.com/Homebrew/homebrew-core/blob/master/Formula/fribidi.rb#L29) for examples using `assert_match`.

Reply:
Also, can you please store the result of `shell_output(...)` and then run `assert_match` on that output, so we don't have to run `step certificate inspect baz.crt` twice.

Reply:
Replaced as many of the `shell_output(...)`s as I could and stored the results so I wouldn't have to run `inspect` multiple times.